### PR TITLE
Added many new options. Volume, preloading, pausing.

### PR DESCRIPTION
**Pull request by z3t0**
Could you also give me write access to this, thanks.
I was thinking we should create an organization to make write access easier.
I am waiting for @maxogden to add me to the other group

### CODE REVIEW

**Comment by z3t0** [comment]
Added the block break sound

**Comment by deathcap** [comment]
This PR can no longer be automatically merged. Could you rebranch and rebase off of master?

**Comment by z3t0** [comment]
thanks for letting me know, il fix it.

**Comment by z3t0** [comment]
@deathcap Do you know where the conflicts are?

File: sfx.js [hunk 1]
```diff
@@ -52,9 +54,10 @@ SfxPlugin.prototype.disable = function() {
   if (this.harvestPlugin) this.harvestPlugin.removeListener('harvested', this.onHarvested);
 };
 
-SfxPlugin.prototype.play = function(name, loop, url) {
+SfxPlugin.prototype.play = function(name, loop, volume, url) {
```
Comment (z3t0) [inline sfx.js]: I'm guessing the conflict is here? If so can you just merge it and fix the conflict because this change was intentional

**Comment by z3t0** [comment]
From what I can see the comment above is the line where the conflict would be? If so that line should be changed

**Comment by deathcap** [comment]
Manually merged and resolved conflicts in https://github.com/deathcap/voxel-sfx/commit/30500c995a0a4332be83b3a0f84fcdb81660671b but in the future could you:

* Branch off of master for making changes in development. To do with git on the command-line: `git checkout -b foo`, where foo is a descriptive branch name (I used `options` in this merge). 
* Commit changes on your branch, submit PR from that branch (so GitHub will say 'wants to merge X commits into deathcap:master from z3t0:options')
* If it can't merge cleanly (GitHub will say so), rebase on top of master, from the command-line: `git rebase master`, assuming `master` is origin/master, fix conflicts

**Comment by z3t0** [comment]
@deathcap Definitely. Its odd because that's exactly what I did... Maybe I
made a mistake along the way. I will make sure there are clean merges next
time, thank you for the tips.

On Tue, Apr 14, 2015 at 7:27 PM, deathcap <notifications@github.com> wrote:

> Manually merged and resolved conflicts in 30500c9
> <https://github.com/deathcap/voxel-sfx/commit/30500c995a0a4332be83b3a0f84fcdb81660671b>
> but in the future could you:
>
>    - Branch off of master for making changes in development. To do with
>    git on the command-line: git checkout -b foo, where foo is a
>    descriptive branch name (I used options in this merge).
>    - Commit changes on your branch, submit PR from that branch (so GitHub
>    will say 'wants to merge X commits into deathcap:master from z3t0:options')
>    - If it can't merge cleanly (GitHub will say so), rebase on top of
>    master, from the command-line: git rebase master, assuming master is
>    origin/master, fix conflicts
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/deathcap/voxel-sfx/pull/10#issuecomment-93136438>.
>



-- 
Thanks,
Rafi Khan

**Comment by deathcap** [comment]
Also regarding:

```javascript
+      var breakSound = this.registry.getProp(blockName, 'breakSound') || 'break/block'
```

where are you getting your "break/block" sound? It's not a sound resource included in https://github.com/deathcap/ProgrammerArt/blob/master/scripts/generate.js#L57 (default art pack for voxel-engine-stackgl / voxel-stitch). Also tested with the BDcraft sound pack from http://bdcraft.net/bdcraft-sounds-pack, it has no sound by this name, closest I could find are:

```
    assets/minecraft/sounds/mob/zombie/woodbreak.ogg   OK
    assets/minecraft/sounds/random/anvil_break.ogg   OK
    assets/minecraft/sounds/random/break.ogg   OK
```

there's also a bunch of per-blocktype dig sounds:

```
    assets/minecraft/sounds/dig/   OK
    assets/minecraft/sounds/dig/grass1.ogg   OK
    assets/minecraft/sounds/dig/grass2.ogg   OK
    assets/minecraft/sounds/dig/grass3.ogg   OK
    assets/minecraft/sounds/dig/grass4.ogg   OK
    assets/minecraft/sounds/dig/grass5.ogg   OK
    assets/minecraft/sounds/dig/grass6.ogg   OK
    assets/minecraft/sounds/dig/grass7.ogg   OK
    assets/minecraft/sounds/dig/grass8.ogg   OK
    assets/minecraft/sounds/dig/gravel1.ogg   OK
    assets/minecraft/sounds/dig/gravel2.ogg   OK
    assets/minecraft/sounds/dig/gravel3.ogg   OK
    assets/minecraft/sounds/dig/gravel4.ogg   OK
    assets/minecraft/sounds/dig/sand1.ogg   OK
    assets/minecraft/sounds/dig/sand2.ogg   OK
    assets/minecraft/sounds/dig/sand3.ogg   OK
    assets/minecraft/sounds/dig/sand4.ogg   OK
    assets/minecraft/sounds/dig/snow1.ogg   OK
    assets/minecraft/sounds/dig/snow2.ogg   OK
    assets/minecraft/sounds/dig/snow3.ogg   OK
    assets/minecraft/sounds/dig/snow4.ogg   OK
    assets/minecraft/sounds/dig/stone1.ogg   OK
    assets/minecraft/sounds/dig/stone2.ogg   OK
    assets/minecraft/sounds/dig/stone3.ogg   OK
    assets/minecraft/sounds/dig/stone4.ogg   OK
    assets/minecraft/sounds/dig/wood.ogg   OK
    assets/minecraft/sounds/dig/wood1.ogg   OK
    assets/minecraft/sounds/dig/wood2.ogg   OK
    assets/minecraft/sounds/dig/wood3.ogg   OK
    assets/minecraft/sounds/dig/wood4.ogg   OK
```

just trying to figure out what would be an appropriate default sound for this action

**Comment by z3t0** [comment]
> where are you getting your "break/block" sound? It's not a sound resource included in

In https://github.com/z3t0/voxelmetaverse/commit/cb4af7bd024c278909975d6649e96d5e66906064 I added the sound to the resource pack. I am a little bit confused as to how I should add things to the resource file so for now its been in the binary zip file.

I only added a simple default break sound which is used if one is not specified. We need to add support for the special ones like grass, wood etc.

**Comment by deathcap** [comment]
Opened a new issue https://github.com/deathcap/voxel-sfx/issues/11 we can discuss there, since this PR is merged

**Pull request merged**